### PR TITLE
Fix _getattr in PrettyDir (#47)

### PR DESCRIPTION
**Pull request by liwt31**
* Improve the logic of the _getattr
* Add format to makefile
* Do reformat using black

### CODE REVIEW

File: pdir/api.py [hunk 1]
```diff
@@ -149,11 +149,8 @@ def _getattr(self):
         """A wrapper around getattr(), handling some exceptions."""
         attrs = {}
         for name in dir(self.obj):
-            try:
-                # Ensures we get descriptor object instead of its return value.
-                attrs[name] = get_dict_attr(self.obj, name)
-            except AttributeError:
```
Comment (laike9m) [inline pdir/api.py]: Why remove this? Not everything is in `__dict__`.

Comment (liwt31) [reply pdir/api.py]: Indeed... I neglected some cases. This should be fixed soon

Comment (laike9m) [reply pdir/api.py]: I don't quite understand what those tests are, but please go ahead.

Comment (laike9m) [reply pdir/api.py]: Actually, seems like the default implementation of `dir` is by looking up `__dict__`
https://github.com/python/cpython/blob/v3.6.4/Objects/typeobject.c#L4469-L4514

So it's probably ok to remove `getattr`. As for user-defined `__dir__`, we'll need to discuss how to deal with it, cause user can put anything(like some names that don't exist) in the return value of `__dir__`.

Comment (liwt31) [reply pdir/api.py]: I add a test with custom `__dir__` and marked it as `xfail`. I believe to deal with custom `__dir__` probably another issue/PR is required.
For the `__dict__` thing, I tried to find a reasonable example but failed. Maybe the current implementation can deal with some tricks around `__dict__`.

File: tests/test_buggy_attrs.py [hunk 2]
```diff
@@ -101,3 +103,21 @@ def p(self):
...
+            self._data = {'foo': 'bar'}
+
+        def __getattr__(self, attr):
+            return self._data.get(attr)
+
+        def __dir__(self):
+            return super(WeirdDir, self).__dir__() + list(self._data.keys())
+
+    pattrs = pdir(WeirdDir()).pattrs
+    assert 'foo' in [pattr.name for pattr in pattrs]
```
Comment (laike9m) [inline tests/test_buggy_attrs.py]: I feel this test is to make sure `pdir`'s behavior is identical to `dir`, so I think you should write
```python
a = 'foo' in [pattr.name for pattr in pattrs]
b = 'foo' in dir(WeirdDir())
assert a == False
assert b == False
```

File: pdir/api.py (hunk 1 above)
Comment (laike9m) [reply pdir/api.py]: Whether you want to remove `getattr` or keep it, I'm ok with both.

**Comment by laike9m** [comment]
Looks good now. You want me to merge it now or you have more things to add?

File: tests/test_buggy_attrs.py [hunk 3]
```diff
@@ -108,7 +108,7 @@ def p(self):
 @pytest.mark.xfail
```
Comment (laike9m) [inline tests/test_buggy_attrs.py]: Remove this.

**Pull request merged**